source/gamemodes/TeamDeathMatch.cpp

```diff
@@ -57,6 +57,33 @@ namespace Duel6 {
         }
     }
 
+    void TeamDeathMatch::initializePlayerPositions(Game &game, std::vector<Player> &players, World &world) const {
+        game.getAppService().getConsole().printLine("...Preparing team players");
+        Level::StartingPositionList startingPositions;
+        world.getLevel().findStartingPositions(startingPositions);
+
+        Int32 layerSpan = startingPositions.size() / teamsCount;
+        Int32 midpoint = layerSpan / 2;

```

Comment:
I think the midpoint may be zero when there are not enough starting positions. And the following code will probably crash with dvision by zero. We already had this problem. It would be nice to handle this edge case as well.